src/screens/Home/Home.js

```diff
@@ -1,25 +1,114 @@
-import { useTheme } from '@react-navigation/native';
-import React from 'react';
-import { Text, View } from 'react-native';
-import { Config } from 'react-native-config';
-import { useSelector } from 'react-redux';
-import { strings } from '@/localization';
-import { getUser } from '@/selectors/UserSelectors';
+/* eslint-disable react-native-a11y/has-valid-accessibility-ignores-invert-colors */
+import React, { useState, useEffect } from 'react';
+import { Text, View, FlatList, Image, ActivityIndicator, TouchableHighlight } from 'react-native';
+import { useQuery } from '@tanstack/react-query';
+import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
+import { useDispatch, useSelector } from 'react-redux';
+import NetInfo from '@react-native-community/netinfo';
 import { styles } from '@/screens/Home/Home.styles';
-import { typography } from '@/theme';
-
-export function Home() {
-  const { colors } = useTheme();
-  const user = useSelector(getUser);
-
-  return (
-    <View style={styles.container}>
-      <Text style={[typography.title, { color: colors.text }]}>
-        {strings.home.message} {user?.username}
-      </Text>
-      <Text style={[typography.text, { color: colors.text }]}>
-        {strings.home.variant} {Config.BUILD_VARIANT}
-      </Text>
+import { IMAGE_URL, routes } from '@/controllers/routes';
+import { NAVIGATION } from '@/constants';
+import { getFav } from '@/selectors/MovieSelectors';
+import { networkService } from '@/networking';
+import { MovieController } from '@/controllers/MovieController';
+import { addIcon } from '@/assets';
+import { saveMovie } from '@/actions/MovieActions';
+
+export function Home({ navigation }) {
+  const movieController = new MovieController(networkService);
+  const [isConnected, setIsConected] = useState(false);
+  const [date, setDate] = useState(new Date());
+  //const { isLoading, data } = useQuery(['allMovies'], movieController.getMovies, {
+  //cacheTime: 0,
+  //});
+
+  const { isLoading, data, isError } = useQuery(['allMovies'], movieController.getMovies);
+
+  const dispatch = useDispatch();
+  const moviesFav = useSelector(getFav);
+
+  useEffect(() => {
+    if (!isError) {
+      setDate(new Date());
+    }
+    const unsubscribe = NetInfo.addEventListener((state) => {
+      setIsConected(state.isConnected);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  const dateDiffInDays = (a, b) => {
+    const _MS_PER_DAY = 1000 * 60 * 60 * 24;
+    const utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());
+    const utc2 = Date.UTC(b.getFullYear(), b.getMonth(), b.getDate());
+    if (Math.floor((utc2 - utc1) / _MS_PER_DAY) < 2) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  if (isLoading) {
+    return <ActivityIndicator />;
+  }
+
+  const onMovie = ({ item }) => (
```

Comment:
Same suggestion as above:

```suggestion
  const MoveItem = ({ item }) => (
```

Reply:
Done!